haskell/private/haskell_impl.bzl

```diff
@@ -320,11 +320,12 @@ def _haskell_binary_common_impl(ctx, is_test):
             ctx.file._bash_runfiles,
             hs.toolchain.tools.hpc,
             binary,
-        ] + mix_runfiles + srcs_runfiles
+        ] + mix_runfiles + srcs_runfiles + java.inputs.to_list()
 
     return [
         hs_info,
         cc_info,
+        java,
```

Comment:
This makes `JavaInteropInfo` part of the public rule interface. I don't think that's necessary to accomplish this. Instead, you could change `java_interop_info` so that it only depends on `deps` and then call it inside `_create_HaskellReplCollectInfo`.

Reply:
Done.